solana/src/gateway.rs

```
@@ -944,53 +945,110 @@ pub(crate) fn query(command: QueryCommands, config: &Config) -> eyre::Result<()>
+        .ok_or_else(|| eyre!("Transaction missing metadata"))?;
 
-    let gateway_id_string = axelar_solana_gateway::id().to_string();
-    let gateway_match_ctx = MatchContext::new(&gateway_id_string);
+    let inner_instructions = meta.inner_instructions.unwrap_or_else(std::vec::Vec::new);
 
-    let invocations = gateway_event_stack::build_program_event_stack(
-        &gateway_match_ctx,
-        &log_messages,
-        gateway_event_stack::parse_gateway_logs,
-    );
-
-    for (i, invocation) in invocations.into_iter().enumerate() {
-        println!("\u{2728} Invocation index [{i}]: ");
-        let events = match invocation {
-            ProgramInvocationState::InProgress(items)
-            | ProgramInvocationState::Failed(items)
-            | ProgramInvocationState::Succeeded(items) => items,
-        };
+    let mut event_count = 0;
 
-        if events.is_empty() {
-            println!("\t\u{1F4EA} No gateway events found");
-        }
+    for (invocation_index, inner_ix_set) in inner_instructions.iter().enumerate() {
+        let mut invocation_events = Vec::new();
 
-        for event in events {
-            print!("\t\u{1F4EC} Event index [{}]: ", event.0);
-            let raw_output = format!("{:#?}", event.1);
-
-            let output = if args.full {
-                &raw_output
-            } else {
-                raw_output.split_once('(').unwrap().0
+        for inner_ix in &inner_ix_set.instructions {
+            let data = match inner_ix {
+                UiInstruction::Compiled(compiled_ix) => {
+                    match base64::engine::general_purpose::STANDARD.decode(&compiled_ix.data) {
+                        Ok(d) => d,
+                        Err(_) => continue,
+                    }
+                }
+                UiInstruction::Parsed(_) => continue,
             };
 
-            println!("{output}");
+            if let Some(event) = parse_gateway_event(&data) {
```

`parse_gateway_event` returns `None` when there's an error parsing the event data. It may hide some potential bugs in the parsing method / events format mismatch in the future. Maybe it's better to return an error from the `parse_gateway_event` and handle it here (e.g. print warn)?